src/Logic.hs

```diff
@@ -706,7 +709,7 @@ pushCandidate (pullRequestId, pullRequest) newHead state =
       -- If something was pushed to the target branch while the candidate was
       -- being tested, try to integrate again and hope that next time the push
       -- succeeds.
-      PushRejected -> tryIntegratePullRequest pullRequestId state
+      PushRejected _why -> tryIntegratePullRequest pullRequestId state
```

Comment:
I like that this re-uses the already existing retry mechanism that exists for regular push failures :+1: